### PR TITLE
Whitelist data-feature & data-field-name html attributes. Resolves #1370

### LINKED ISSUE #1370: Hitting error when saving changes through dashboard
__Is this a bug, question or feature request?__
Bug

__Describe the issue you encountered:__
Hitting javascript error in console whenever I try to change and save settings through the dashboard. Chrome 74.0.3729.169.

__Current WordPress version?__
v4.8.3

__Current ElasticPress version?__
v3.0.3

__Current Elasticsearch version?__
v6.5.4

__Where do you host your Elasticsearch server?__
bonsai.io

__What ElasticPress features do you have currently active?__
Default installation

__Steps to reproduce:__

1. Install the plugin through Wordpress Plugins. Update settings to include Elasticsearch server. Start indexing. All fine up to this step.
2. Go to Elasticsearch dashboard. Turn off Post Search, click Save, and it won't save successfully. The error is available in Javascript console.

__Screenshots, if needed:__
https://ibb.co/zRXsWVy
https://ibb.co/rwsBrwT

### PR DESCRIPTION
**Pull request by mnazwan**
### Applicable Issues
Resolves #1370

### CODE REVIEW

File: includes/dashboard.php [hunk 1]
```diff
@@ -65,18 +65,19 @@ function filter_allowed_html( $allowedtags, $context ) {
...
-			'title'       => true,
-			'name'        => true,
-			'placeholder' => '',
+			'type'            => true,
+			'checked'         => true,
+			'selected'        => true,
+			'disabled'        => true,
+			'value'           => true,
+			'class'           => true,
+			'data-*'          => true,
```
Comment (andersrehn) [inline includes/dashboard.php]: I get the feeling that `data-*` does not work as expected in at least a wordpress `4.9.10` version. 

Expected behaviour is to allow all data- tags. 
Actual result is that you must for each given data key allow them.

File: includes/dashboard.php [hunk 2]
```diff
@@ -94,6 +95,8 @@ function filter_allowed_html( $allowedtags, $context ) {
 			'target'         => true,
 		];
 
+		$ep_tags['a']['data-feature'] = true;
```
Comment (andersrehn) [inline includes/dashboard.php]: This does indeed solve the issue that the console is broken. However the dashboard does not save any data even if the UI tells you it did. 

If we add `$ep_tags['input']['data-field-name'] = true;` to the line below, the data will be saved as well.

Comment (mnazwan) [reply includes/dashboard.php]: Changes required for ```<input>``` already covered in default whitelisted attributes, Line 75.

File: includes/dashboard.php (hunk 1 above)
Comment (mnazwan) [reply includes/dashboard.php]: Agree. This is just a quick fix.

**Comment by tlovett1** [comment]
Pushed a commit to develop very similar to this but added a few more attributes. Thanks guys.

**Pull request closed without merging**